supabase/migrations/20250523191212_assignment-eval-settings.sql

```diff
@@ -0,0 +1,245 @@
+CREATE TABLE "public"."self_review_settings" (
+    "id" bigint generated by default as identity not null,
+    "enabled" boolean not null default false,
+    "deadline_offset" bigint,
+    "allow_early" boolean,
+    "class_id" bigint not null 
+);
+
+ALTER TABLE "public"."self_review_settings" ENABLE ROW LEVEL SECURITY;
+
+CREATE POLICY "anyone in the course can view self review settings" 
+ON "public"."self_review_settings"
+AS PERMISSIVE
+FOR SELECT
+USING (
+    authorizeforclassinstructor(class_id) OR authorizeforclassgrader(class_id) 
+    OR authorizeforclass(class_id)
+);
+
+CREATE POLICY "instructors can insert self review settings" 
+ON "public"."self_review_settings"
+AS PERMISSIVE
+FOR INSERT
+WITH CHECK (
+    authorizeforclassinstructor(class_id)
+);
+
+CREATE POLICY "instructors can update self review settings" 
+ON "public"."self_review_settings"
+AS PERMISSIVE
+FOR UPDATE
+USING (
+    authorizeforclassinstructor(class_id) 
+);
+
+CREATE POLICY "instructors can delete review settings" 
+ON "public"."self_review_settings"
+AS PERMISSIVE
+FOR DELETE
+USING (
+    authorizeforclassinstructor(class_id)
+);
+
+ALTER TABLE ONLY "public"."self_review_settings"
+    ADD CONSTRAINT "self_review_settings_pkey" PRIMARY KEY ("id");
+
+ALTER TABLE "public"."self_review_settings"
+    ADD CONSTRAINT "self_review_settings_class_fkey" FOREIGN KEY ("class_id") REFERENCES "public"."classes"("id");
+
+ALTER TABLE "public"."assignments" ADD COLUMN "self_review_setting_id" bigint NOT NULL;
+
+ALTER TABLE "public"."assignments"
+    ADD CONSTRAINT "assignments_self_review_setting_fkey" FOREIGN KEY ("self_review_setting_id") REFERENCES "public"."self_review_settings"("id");
+
+CREATE POLICY "Students can give themselves negative deadline exceptions"
+ON "public"."assignment_due_date_exceptions"
+AS PERMISSIVE 
+FOR INSERT
+WITH CHECK (
+   "hours" < 0 
```

Comment:
I can't easily tell how all of the policies interact, but: is there something else that restricts students to only be able to insert negative extensions for themselves?

Reply:
Sorry, what I meant was: I think that the check should also enforce `authorizeforprofile(student_id) or authorizeforassignmentgroup(assignment_group_id)`, to prevent a student from entering a negative extension for another student (unless there is something else that is enforcing this, but I didn't think so)